### PR TITLE
Logic Implementation

### PR DESCRIPTION
**Pull request by Jarfsoft**
**Feature added:**

- Logic implementation
- Styles to buttons
- Styles to Display

### CODE REVIEW

File: src/Components/App.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import React, { useState } from 'react';
 import Display from './Display';
 import ButtonPanel from './ButtonPanel';
 import calculate from '../Logic/calculate';
+import './App.css';
 
 function App() {
   const [data, setData] = useState({
```
Comment (arslanbisharat) [inline src/Components/App.js]: - [x] App is converted to a stateful component using class ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/app/stateful.png))
- [x] App has state initialized in constructor ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/app/stateful.png))
- [x] App has `handleClick` implemented in constructor

```javascript
constructor(props) {
  super(props);
  this.state = {
      total: null,
      next: null,
      operation: null,
  };
  }

  handleClick = (buttonName) => {
  this.setState(calculate(this.state, buttonName));
  }
}
```

- [ ] App passes `handleClick` as a `clickHandler` property to `ButtonPanel` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/app/passHandleClick.png))
- [ ] App passes the `total` or `next` value from state to `Display` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/app/passToatalToDisplay.png))

File: src/Components/Button.js [hunk 2]
```diff
@@ -14,6 +21,7 @@ Button.propTypes = {
   name: PropTypes.string.isRequired,
   press: PropTypes.func.isRequired,
   key: PropTypes.string.isRequired,
+  fr: PropTypes.string.isRequired,
 };
 
 export default Button;
```
Comment (arslanbisharat) [inline src/Components/Button.js:27]: - [x] Button attaches `handleClick` to `onClick` event on Button

File: src/Components/ButtonPanel.js [hunk 3]
```diff
@@ -2,36 +2,47 @@ import React, { Component } from 'react';
...
+        (btn) => <Button fr={fraction} key={btn} press={sIH} name={btn} />,
+      );
+    }
+
+    return (
+      <>
+        <Button fr="1" key="0" press={sIH} name="0" />
+        <Button fr="1" key="." press={sIH} name="." />
+        <Button fr="0" key="=" press={sIH} name="=" />
+      </>
```
Comment (arslanbisharat) [inline src/Components/ButtonPanel.js:22]: - [x] ButtonPanel implements `handleClick` function ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/panel/implementHandleClick.png))

```javascript
handleClick = (buttonName) => {
  this.props.clickHandler(buttonName);
};
```

- [x] ButtonPanel passes `handleClick` as `clickHandler` props to each `Button` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/panel/passToButton.png))

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi @Jarfsoft, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/Components/App.js [hunk 4]
```diff
@@ -1,34 +1,50 @@
...
+  }
+  // const [data, setData] = useState({
+  //  total: null,
+  //  next: null,
+  //  operation: null,
+  // });
+
+  // const [display, setDisplay] = useState(
+  //  <Display result={data} />,
+  // );
```
Comment (afanimartin) [inline src/Components/App.js]: - Remove this comment.

**Review by afanimartin — CHANGES_REQUESTED** [review]
## Changes Required 🙏🏿 🙏🏿 🙏🏿 
Nice work making the requested changes.
`App` component turned into a class component as requested. Well done.
### Change to be made 🙅🏿 🙅🏿 🙅🏿 
- Remove comments from your code.

### Martin Afani
#### Slack: whiz25

**Review by Torres-ssf — APPROVED** [review]
### Milestone Approved :+1: 
Hello @Jarfsoft 
Nice job on this milestone
Good luck with the next one

Happy coding :computer: 
**Sérgio Torres**

**Pull request merged**